src/c++/perf_analyzer/genai-pa/tests/test_cli.py

```diff
@@ -24,14 +24,68 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+import io
+import sys
+
 import pytest
+from genai_pa import parser
 from genai_pa.main import run
 
 
-# NOTE: Placeholder
-class TestHelp:
-    @pytest.mark.parametrize("arg", ["-h", "--help"])
-    def test_help(self, arg):
-        args = [arg]
-        with pytest.raises(SystemExit):
-            run(args)
+class TestCLIArguments:
+    @pytest.mark.parametrize(
+        "arg, expected_output",
+        [
+            (["-h"], "CLI to profile LLMs and Generative AI models with Perf Analyzer"),
+            (
+                ["--help"],
+                "CLI to profile LLMs and Generative AI models with Perf Analyzer",
+            ),
+        ],
+    )
+    def test_help_arguments_output_and_exit(self, arg, expected_output, capsys):
+        with pytest.raises(SystemExit) as excinfo:
+            _ = parser.parse_args(arg)
+
+        # Check that the exit was successful
+        assert excinfo.value.code == 0
+
+        # Capture that the correct message was displayed
+        captured = capsys.readouterr()
+        assert expected_output in captured.out
+
+    @pytest.mark.parametrize(
+        "arg, expected_output",
+        [
+            (["-b", "2"], "batch_size=2"),
+            (["--batch-size", "2"], "batch_size=2"),
+            (["--concurrency", "3"], "concurrency_range='3'"),
+            (["--max-threads", "4"], "max_threads=4"),
+            (
+                ["--profile-export-file", "text.txt"],
+                "profile_export_file=PosixPath('text.txt')",
+            ),
+            (["--request-rate", "1.5"], "request_rate_range='1.5'"),
+            (["--service-kind", "triton"], "service_kind='triton'"),
+            (["--service-kind", "openai"], "service_kind='openai'"),
+            # TODO: Remove streaming from implementation. It is invalid with HTTP.
+            # (["--streaming"], "Streaming=True"),
+            (["--version"], "version=True"),
+            (["-u", "test_url"], "u='test_url'"),
+            (["--url", "test_url"], "u='test_url'"),
+        ],
+    )
+    def test_arguments_output(self, arg, expected_output, capsys):
+        combined_args = ["--model", "test_model"] + arg
+        _ = parser.parse_args(combined_args)
+
+        # Capture that the correct message was displayed
+        captured = capsys.readouterr()
```

Comment:
Still ramping on this. What is being run here to give output?
`parser.parse_args(combined_args)` doesn't print anything, so I am not 100% sure where the I/O checking comes in. Totally possible I don't fully grok the way capsys works yet.

Reply:
You're right. This should not be passing. I'll re-run it and try to see what's happening, because this is actually not the behavior we want. If anything, we should confirm that there is nothing being printed.

Reply:
If you run genai-pa, you'll see that it does print the namespace: "Namespace(model='opt125m', batch_size=1, max_threads=16, profile_export_file=PosixPath('profile_export.json'), service_kind='triton', streaming=False, version=False, u='localhost:8001', func=<function handler at 0x7faed0cc93f0>)". That's because there is no formatter specified.

I can change the test, since it's testing current behavior which it sounds like is not the behavior we want in our implementation. We need to fix that on the implementation side as well.

Reply:
Based on conversations in Slack, the tests have been updated to not rely on standard output. This test will fail until the implementation is updated to not print the namespace. Assuming there is no more feedback though, I think this would be fine to merge in the ethos of having the unit tests done before implementation is complete. Once the implementation is updated, they'll all pass.

CC: @rmccorm4 for advice on not printing the namespace. Looking at our other repos, this seems to be accepted practice. I think you can probably work around the logic, but ArgumentParser's arg_parse prints the namespace unless you use its formatter to turn off the printing then manually handle the `-h/--help` commands and errors.

Reply:
It was my bad. 
Left a debug print in place over in the parser.
if you remove the line in parser.prune_args(), it will behave the way we expect